### PR TITLE
Fix LastCreateScene value when deleting scenes

### PR DESCRIPTION
**Pull request by Popov72**
I'm not sure it is a fix, maybe it is expected that `LastCreatedScene` is null when we delete the last created scene, even if there are still some other scenes alive...

However, personally I would expect to have the "last scene not disposed" in `LastCreatedScene`, so let's tell what you think.

### CODE REVIEW

File: src/scene.ts [hunk 1]
```diff
@@ -4480,6 +4480,14 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
         this.afterRender = null;
         this.metadata = null;
 
+        if (EngineStore._LastCreatedScene === this) {
```
Comment (deltakosh) [inline src/scene.ts:4483]: OVerall I'm good with this change but you need to remove the next block of code (duplicate)

Comment (Popov72) [reply src/scene.ts:4483]: I already did it before you comment ;)

File: src/scene.ts [hunk 2]
```diff
@@ -4481,7 +4481,11 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
         this.metadata = null;
 
         if (EngineStore._LastCreatedScene === this) {
-            EngineStore._LastCreatedScene = null;
+            if (this._engine.scenes.length > 1) {
```
Comment (RaananW) [inline src/scene.ts:4484]: Wouldn't this entire code fit better after removing the scene from `engine.scenes` in line 485?

**Review by sebavan — APPROVED** [review]

**Comment by Popov72** [comment]
The error in **Module Tests** is:
image

**Review by RaananW — APPROVED** [review]
Nit-picky - the else is actually not needed. Simply set it to be the last element of the array. If there isn't any, it'll be undefined negative index will return undefined as well)

**Comment by Popov72** [comment]
Indeed but `_LastCreatedScene` is not declared as accepting to be `undefined`:
```javascript
public static _LastCreatedScene: Nullable<Scene> = null;
```
I find it strange that:
```javascript
EngineStore._LastCreatedScene = undefined;
```
is (rightly) not allowed by Typescript but that this code does not raise any warning/error:
```javascript
EngineStore._LastCreatedScene = this._engine.scenes[this._engine.scenes.length - 1];
```

**Pull request merged**